src/data/InstanceDhisRepository.ts

```diff
@@ -680,12 +685,13 @@ export class InstanceDhisRepository implements InstanceRepository {
                         orgUnit,
                         period: moment(occurredAt).format("YYYY-MM-DD"),
                         attribute: attributeOptionCombo,
-                        coordinate: geometry
-                            ? {
-                                  longitude: geometry.coordinates[0]?.toString() ?? "",
-                                  latitude: geometry.coordinates[1]?.toString() ?? "",
-                              }
-                            : coordinate,
+                        coordinate:
+                            geometry && geometry.type === "Point"
+                                ? {
+                                      longitude: geometry.coordinates[0]?.toString() ?? "",
+                                      latitude: geometry.coordinates[1]?.toString() ?? "",
+                                  }
```

Comment:
Not your code, but now that we are refactoring this, do we need `?`. TS says it's a Coordinate, so we have [a, b] for sure.